library/azure_rm_hdinsightcluster.py

```diff
@@ -0,0 +1,482 @@
+        return self.results
+
+    def create_update_cluster(self):
+        '''
+        Creates or updates Cluster with the specified configuration.
+
+        :return: deserialized Cluster instance state dictionary
+        '''
+        self.log("Creating / Updating the Cluster instance {0}".format(self.name))
+
+        try:
+            if self.to_do == Actions.Create:
+                response = self.mgmt_client.clusters.create(resource_group_name=self.resource_group,
+                                                            cluster_name=self.name,
+                                                            parameters=self.parameters)
+            else:
+                response = self.mgmt_client.clusters.update(resource_group_name=self.resource_group,
+                                                            cluster_name=self.name,
+                                                            parameters=self.parameters)
+            if isinstance(response, LROPoller):
+                response = self.get_poller_result(response)
+
+        except CloudError as exc:
+            self.log('Error attempting to create the Cluster instance.')
+            self.fail("Error creating the Cluster instance: {0}".format(str(exc)))
+        return response.as_dict()
+
+    def delete_cluster(self):
+        '''
+        Deletes specified Cluster instance in the specified subscription and resource group.
+
+        :return: True
+        '''
+        self.log("Deleting the Cluster instance {0}".format(self.name))
+        try:
+            response = self.mgmt_client.clusters.delete(resource_group_name=self.resource_group,
+                                                        cluster_name=self.name)
+        except CloudError as e:
+            self.log('Error attempting to delete the Cluster instance.')
+            self.fail("Error deleting the Cluster instance: {0}".format(str(e)))
+
+        return True
+
+    def get_cluster(self):
+        '''
+        Gets the properties of the specified Cluster.
+
+        :return: deserialized Cluster instance state dictionary
+        '''
+        self.log("Checking if the Cluster instance {0} is present".format(self.name))
+        found = False
+        try:
+            response = self.mgmt_client.clusters.get(resource_group_name=self.resource_group,
+                                                     cluster_name=self.name)
+            found = True
+            self.log("Response : {0}".format(response))
+            self.log("Cluster instance : {0} found".format(response.name))
+        except Exception as e:
+            self.log('Did not find the Cluster instance.')
+        if found is True:
+            return response.as_dict()
+
+        return False
+
+    def format_item(self, item):
+        if not item:
+            return {}    
+        d = {
+            'id': item['id']
+        }
+        return d
+
+
+def _snake_to_camel(snake, capitalize_first=False):
```

Comment:
from ansible.module_utils.common.dict_transformations import (
    _camel_to_snake, _snake_to_camel,
).

If this not in role, how about add it as  common utilities?